packages/plugin-ext/src/plugin/plugin-manager.ts

```diff
@@ -119,28 +126,58 @@ export class PluginManagerExtImpl implements PluginManagerExt, PluginManager {
             return;
         }
         this.activatedPlugins.delete(pluginId);
-        this.stopPlugin(plugin);
+        return this.stopPlugin(pluginId, plugin);
+    }
+
+    async terminate(): Promise<void> {
+        return this.stopAll({ terminating: true });
     }
 
-    protected stopAll(): void {
-        this.activatedPlugins.forEach(plugin => this.stopPlugin(plugin));
+    protected async stopAll(options: StopOptions = { terminating: false }): Promise<void> {
+        const promises = [];
+        for (const [id, plugin] of this.activatedPlugins) {
+            promises.push(this.stopPlugin(id, plugin, options));
+        }
 
         this.registry.clear();
         this.loadedPlugins.clear();
         this.activatedPlugins.clear();
         this.pluginActivationPromises.clear();
         this.pluginContextsMap.clear();
+        await Promise.all(promises);
     }
 
-    protected stopPlugin(plugin: ActivatedPlugin): void {
+    protected async stopPlugin(id: string, plugin: ActivatedPlugin, options: StopOptions = { terminating: false }): Promise<void> {
+        let result;
         if (plugin.stopFn) {
-            plugin.stopFn();
+            try {
+                result = plugin.stopFn();
+            } catch (e) {
+                if (!options.terminating) {
+                    console.error(`[${id}]: failed to stop:`, e);
```

Comment:
there is a logger in that class, probably better to use it.

Reply:
I don't think the plugin host process has access to the logger. But maybe i'm overlooking something.

Reply:
I make comment on the wrong console.error line

Reply:
acknowledged :)